fv3gfs/physics/stencils/update_dwind_phys.py

```diff
@@ -1,699 +1,566 @@
-def update_dwind_phys(data):
-    dt5 = 0.5 * data["dt"]
-    im2 = int((data["npx"] - 1) / 2) + 2
-    jm2 = int((data["npy"] - 1) / 2) + 2
-    # assert grid_type < 3
-    shape = data["u"].shape
-    ue_1 = make_storage_from_shape(shape)
-    ue_2 = make_storage_from_shape(shape)
-    ue_3 = make_storage_from_shape(shape)
-    ut_1 = make_storage_from_shape(shape)
-    ut_2 = make_storage_from_shape(shape)
-    ut_3 = make_storage_from_shape(shape)
-    ve_1 = make_storage_from_shape(shape)
-    ve_2 = make_storage_from_shape(shape)
-    ve_3 = make_storage_from_shape(shape)
-    vt_1 = make_storage_from_shape(shape)
-    vt_2 = make_storage_from_shape(shape)
-    vt_3 = make_storage_from_shape(shape)
-    update_dwind_prep_stencil(
-        data["u_dt"],
-        data["v_dt"],
-        data["vlon1"],
-        data["vlon2"],
-        data["vlon3"],
-        data["vlat1"],
-        data["vlat2"],
-        data["vlat3"],
-        ue_1,
-        ue_2,
-        ue_3,
-        ve_1,
-        ve_2,
-        ve_3,
-        origin=(HALO - 1, HALO - 1, 0),
-        domain=(data["nic"] + 2, data["njc"] + 2, data["npz"]),
-    )
-
-    # if grid.west_edge
-    if data["is"] == HALO:
-        if data["js"] <= jm2:
-            je_lower = min(jm2, data["je"])
-            origin_lower = (HALO, HALO, 0)
-            domain_lower = (1, je_lower - data["js"] + 1, data["npz"])
-            if domain_lower[1] > 0:
-                update_dwind_y_edge_south_stencil(
-                    ve_1,
-                    ve_2,
-                    ve_3,
-                    vt_1,
-                    vt_2,
-                    vt_3,
-                    data["edge_vect_w"],
-                    origin=origin_lower,
-                    domain=domain_lower,
+class AGrid2DGridPhysics:
+    """
+    Fortran name is update_dwinds_phys
+    """
+
+    def __init__(self, grid, namelist):
+        self.grid = grid
+        self.namelist = namelist
+        self._dt5 = 0.5 * self.namelist.dt_atmos
+        self._im2 = int((self.grid.npx - 1) / 2) + 2
+        self._jm2 = int((self.grid.npy - 1) / 2) + 2
+        shape = self.grid.grid_indexing.max_shape
+        self._ue_1 = utils.make_storage_from_shape(shape, init=True)
+        self._ue_2 = utils.make_storage_from_shape(shape, init=True)
+        self._ue_3 = utils.make_storage_from_shape(shape, init=True)
+        self._ut_1 = utils.make_storage_from_shape(shape, init=True)
+        self._ut_2 = utils.make_storage_from_shape(shape, init=True)
+        self._ut_3 = utils.make_storage_from_shape(shape, init=True)
+        self._ve_1 = utils.make_storage_from_shape(shape, init=True)
+        self._ve_2 = utils.make_storage_from_shape(shape, init=True)
+        self._ve_3 = utils.make_storage_from_shape(shape, init=True)
+        self._vt_1 = utils.make_storage_from_shape(shape, init=True)
+        self._vt_2 = utils.make_storage_from_shape(shape, init=True)
+        self._vt_3 = utils.make_storage_from_shape(shape, init=True)
```

Comment:
The numbers are just vector components, so if you don't like _1, _2, _3, could instead us _i, _j, _k or _x, _y, _z. If it's the ue, ut that is unappealing, I think it might mean cell edge x-wind tendency. It looks like the ut, vt variables are simply temporaries to assign tile edges. We actually shouldn't need the the ut/vt variables or the copy stencils -- except for race conditions. We can't write a variable after reading it with an offset. I wonder if something like `ve_1 = edge_vect * ve_1[0, 1, 0] + (1.0 - edge_vect) * ve_1` counts as that situation. Anyways, you can leave for now, but we might want to revisit that in another iteration.

Reply:
another iteration could also put this stuff in regions if we can do so without race conditions

Reply:
Okay, in this case I will leave them as they are. For the number, I was more concerned about names like: `self._update_dwind_y_edge_south_stencil1`

Reply:
Ah I see. they can be named then based on what they refer to, but should go away with a regions refactor